### PR TITLE
Migrate watcher ring buffer and dprint buffers to dev_mem_map

### PR DESCRIPTION
**Pull request by pgkeller**
### Ticket
#4984 

### Problem description
There are 2 versions of the memory map, this is work towards unifying them

### What's changed
watcher ring buffer and dprint buffers are no longer in common_runtime_address_map.h, now in dev_msgs.h

### Checklist
- [ ] Post commit CI passes
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tt_metal/hostdevcommon/common_runtime_address_map.h [hunk 1]
```diff
@@ -77,24 +77,7 @@ constexpr static std::uint32_t SEMAPHORE_BASE = PROFILER_L1_BUFFER_CONTROL + PRO
...
-constexpr static std::uint32_t PRINT_BUFFER_BR = PRINT_BUFFER_T2 + PRINT_BUFFER_SIZE; // BRISC
-constexpr static std::uint32_t PRINT_BUFFER_IDLE_ER = PRINT_BUFFER_START; // Idle ERISC
-
-// Debug ring buffer, shared between all cores
-constexpr static std::uint32_t RING_BUFFER_ADDR = PRINT_BUFFER_START + PRINT_BUFFER_MAX_SIZE;
-constexpr static std::uint32_t RING_BUFFER_SIZE = 128;
-
-constexpr static std::uint32_t L1_UNRESERVED_BASE = (((RING_BUFFER_ADDR + RING_BUFFER_SIZE) - 1) | (DRAM_ALIGNMENT - 1)) + 1;
+constexpr static std::uint32_t L1_UNRESERVED_BASE = ((SEMAPHORE_BASE + SEMAPHORE_SIZE - 1) | (DRAM_ALIGNMENT - 1)) + 1;
 constexpr static std::uint32_t ERISC_L1_UNRESERVED_BASE = L1_UNRESERVED_BASE; // Start of unreserved space
```
Comment (aliuTT) [inline tt_metal/hostdevcommon/common_runtime_address_map.h:81]: We should probably get rid of `ERISC_L1_UNRESERVED_BASE`? I think it's used for idle erisc cores.

**Review by aliuTT — APPROVED** [review]

File: tt_metal/hw/inc/blackhole/dev_mem_map.h [hunk 2]
```diff
@@ -50,10 +50,10 @@
 
 #define MEM_BOOT_CODE_BASE 0
 #define MEM_MAILBOX_BASE 16
-#define MEM_MAILBOX_END (MEM_MAILBOX_BASE + 256)
+#define MEM_MAILBOX_END (MEM_MAILBOX_BASE + 1312)
```
Comment (abhullar-tt) [inline tt_metal/hw/inc/blackhole/dev_mem_map.h:53]: is 1312 size of MAILBOX?

File: tt_metal/hw/inc/blackhole/eth_l1_address_map.h [hunk 3]
```diff
@@ -56,15 +55,12 @@ struct address_map {
   // erisc early exit functionality re-uses mailboxes_t::ncrisc_halt_msg_t::stack_save memory
   static constexpr std::int32_t ERISC_MEM_MAILBOX_STACK_SAVE = ERISC_MEM_MAILBOX_BASE + 4;
 
-  static constexpr std::int32_t ERISC_RING_BUFFER_ADDR = ERISC_MEM_MAILBOX_BASE + 256 + 16;
-  static constexpr std::int32_t PRINT_BUFFER_ER = ERISC_RING_BUFFER_ADDR + RING_BUFFER_SIZE;
+  static constexpr std::int32_t PRINT_BUFFER_ER = ERISC_MEM_MAILBOX_BASE + 288 + 16;
```
Comment (abhullar-tt) [inline tt_metal/hw/inc/blackhole/eth_l1_address_map.h:58]: 288 here?

**Review by tt-dma — APPROVED** [review]
Looks good to me outside of comments already left by others

File: tt_metal/hw/inc/blackhole/eth_l1_address_map.h (hunk 3 above)
Comment (pgkeller) [reply tt_metal/hw/inc/blackhole/eth_l1_address_map.h:58]: Yeah, all the values for sizes are hard coded but asserted.  I have (at least) 2 more passes to make over this, removing profiler and splitting kernels/firmware.  I'll be sure to remove the magic when the dust settles and before I move onto the next thing

**Pull request merged**